cdap-docs/build.sh

```diff
@@ -489,36 +496,112 @@ function build_license_dependency_pdfs() {
 }
 
 function build_standalone() {
-  local add_artifacts=""
+  local errors
+  local check="0"
+  local additional_artifacts_path="/tmp/${ADDITIONAL_ARTIFACTS}"
+  local add_artifacts="-Dadditional.artifacts.dir=${additional_artifacts_path}"
+  rm -rf ${additional_artifacts_path}
+  mkdir -p ${additional_artifacts_path}
+  echo "Cleaned directory ${additional_artifacts_path}"
   if [ -d ${HYDRATOR_PLUGINS_PATH} ]; then
-    build_hydrator_plugins
-    local errors=$?
-    if [ "${errors}" == "0" ]; then
-      add_artifacts="-Dadditional.artifacts.dir=${HYDRATOR_PLUGINS_PATH}"
+    build_hydrator_plugins ${additional_artifacts_path}
+    errors=$?
+    if [ "${errors}" != "0" ]; then
+      echo "Error building Hydrator plugins: ${errors}"
+      check=1
     fi
   else
     echo "No HYDRATOR_PLUGINS_PATH at ${HYDRATOR_PLUGINS_PATH}"
+    check=1
+  fi
+  check_if_continue ${check}
+  check="0"
+  local add_apps=""
+  if [ -d ${TRACKER_APP_PATH} ]; then
+    build_tracker_app ${additional_artifacts_path}
+    errors=$?
+    if [ "${errors}" != "0" ]; then
+      echo "Error building Tracker app: ${errors}"
+      check=1
+    fi
+  else
+    echo "No TRACKER_APP_PATH at ${TRACKER_APP_PATH}"
+    check=1
   fi
+  check_if_continue ${check}
+  echo "Using additional artifacts: ${add_artifacts}"
   set_mvn_environment
   MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean package \
   -pl cdap-standalone,cdap-app-templates/cdap-etl,cdap-app-templates/cdap-data-quality,cdap-examples \
   -am -amd -DskipTests -P examples,templates,dist,release,unit-tests ${add_artifacts}
 }
 
 function build_hydrator_plugins() {
+  local additional_artifacts_path=$1
   local errors=0
   set_mvn_environment
   if [ -d ${HYDRATOR_PLUGINS_PATH} ]; then
     cd ${HYDRATOR_PLUGINS_PATH}
     echo "HYDRATOR_PLUGINS_PATH: ${HYDRATOR_PLUGINS_PATH}"
     mvn clean package -DskipTests
     errors=$?
+    if [ "${errors}" == "0" ]; then
+      echo "Copying Hydrator to ${additional_artifacts_path}"
+      local artifacts="${additional_artifacts_path}/${HYDRATOR_PLUGINS}/${TARGET}"
+      if [ -d ${additional_artifacts_path} ]; then
+        rm -rf ${artifacts}
+        mkdir -p ${artifacts}
+        find ${HYDRATOR_PLUGINS_PATH} \( -name '*.jar' -o -name '*.json' \) -print0 | xargs -I{} -0 cp {} ${artifacts}
+        errors=$?
+        echo "Copied Hydrator to ${artifacts}"
+      fi
+    fi
   else
     echo "No HYDRATOR_PLUGINS_PATH at ${HYDRATOR_PLUGINS_PATH}"
   fi
   return ${errors}
 }
-  
+
+function build_tracker_app() {
+  local additional_artifacts_path=$1
```

Comment:
This function is a copy of the above. There's no reason for the local copy of `additional_artifacts_path` since this is a shell script and it is a constant. Set it once for the entire script. Instead, the argument to this function should be the source path to build. Maybe rename it to "build_external_app" or something to match it's now generic use.